lib/gui/os/dialog.js

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const _ = require('lodash');
+const electron = require('electron');
+const errors = require('../../shared/errors');
+const supportedFormats = require('../../shared/supported-formats');
+
+/**
+ * @summary Current renderer BrowserWindow instance
+ * @type {Object}
+ * @private
+ */
+const currentWindow = electron.remote.getCurrentWindow();
+
+/**
+ * @summary Open an image selection dialog
+ * @function
+ * @public
+ *
+ * @description
+ * Notice that by image, we mean *.img/*.iso/*.zip/etc files.
+ *
+ * @fulfil {Object} - selected image
+ * @returns {Promise};
+ *
+ * @example
+ * osDialog.selectImage().then((image) => {
```

Comment:
It makes it more susceptible to a name clash like `exception` so I think let's keep it for now.